pkg/localcontroller/managers/incrementallearning/incrementallearningjob.go

```diff
@@ -139,53 +144,65 @@ func (im *Manager) Start() error {
 }
 
 // trainTask starts training task
-func (im *Manager) trainTask(job *Job, currentRound int) error {
+func (im *Manager) trainTask(job *Job) error {
 	jobConfig := job.JobConfig
 
 	latestCond := im.getLatestCondition(job)
 	jobStage := latestCond.Stage
 	currentType := latestCond.Type
 
 	if currentType == sednav1.ILJobStageCondWaiting {
-		if job.Dataset == nil {
+		var err error
+
+		err = im.loadDataset(job)
+		if err != nil || jobConfig.Dataset == nil || jobConfig.Dataset.DataSource == nil {
+			return fmt.Errorf("job(name=%s) failed to load dataset, and waiting it: %v",
+				jobConfig.UniqueIdentifier, err)
+		}
+
+		if jobConfig.Dataset == nil || jobConfig.Dataset.DataSource == nil {
 			return fmt.Errorf("job(name=%s) dataset not ready", jobConfig.UniqueIdentifier)
 		}
 
-		err := im.loadTrainModel(job)
+		err = im.loadTrainModel(job)
 		if err != nil {
 			return fmt.Errorf("failed to sync train model, and waiting it: %v", err)
 		}
 
-		if currentRound < jobConfig.Rounds {
-			currentRound = jobConfig.Rounds
-			initTriggerStatus(jobConfig)
-		}
-	}
+		initTriggerStatus(jobConfig)
 
-	if currentType == sednav1.ILJobStageCondWaiting && jobConfig.TrainTriggerStatus == TriggerReadyStatus {
-		payload, ok, err := im.triggerTrainTask(job)
-		if !ok {
-			return nil
-		}
+		if jobConfig.TrainTriggerStatus == TriggerReadyStatus {
+			payload, ok, err := im.triggerTrainTask(job)
+			if !ok {
+				return nil
+			}
 
-		if err != nil {
-			klog.Errorf("job(name=%s) complete the %sing phase triggering task failed, error: %v",
-				jobConfig.UniqueIdentifier, jobStage, err)
-			return err
-		}
+			if err != nil {
+				klog.Errorf("job(name=%s) failed to complete the %sing phase triggering task: %v",
+					jobConfig.UniqueIdentifier, jobStage, err)
+				job.JobConfig.Rounds--
+				return err
+			}
 
-		err = im.Client.WriteMessage(payload, job.getHeader())
-		if err != nil {
-			klog.Errorf("job(name=%s) failed to write message: %v",
-				jobConfig.UniqueIdentifier, err)
-			return err
-		}
+			err = im.Client.WriteMessage(payload, job.getHeader())
+			if err != nil {
+				klog.Errorf("job(name=%s) failed to write message: %v", jobConfig.UniqueIdentifier, err)
+				job.JobConfig.Rounds--
+				return err
+			}
 
-		jobConfig.TrainTriggerStatus = TriggerCompletedStatus
-		jobConfig.Rounds++
-		forwardSamples(jobConfig, jobStage)
-		klog.Infof("job(name=%s) complete the %sing phase triggering task successfully",
-			jobConfig.UniqueIdentifier, jobStage)
+			forwardSamples(jobConfig, jobStage)
+
+			err = im.saveJobToDB(job)
+			if err != nil {
+				klog.Errorf("job(name=%s) failed to update job to db, error: %v",
+					jobConfig.UniqueIdentifier, err)
```

Comment:
no `return` here instead of continuing since an error has been happend and logged? if intentional, add a comment here

Reply:
like `// continue anyway`

Reply:
fix